modules/azure/function_app_linux_managed_identity/outputs.tf

```diff
@@ -13,3 +13,7 @@ output "id" {
 output "aad_application_name" {
   value = var.managed_identity_provider.existing == null ? "api://${var.managed_identity_provider.create.application_name}" : null
 }
+
+output "api_audience" {
+  value = tolist(tolist(tolist(data.azurerm_linux_function_app.function_app.auth_settings)[0].active_directory)[0].allowed_audiences)[0]
```

Comment:
1. Are all these `tolist` really needed, feels to me like a lot.
2. `azurerm_linux_function_app.function_app` is not a `data` but a `resource`.
3. Is there ever going to be any data in here? 
I can't find `auth_settings` anywhere in the main file, so it looks to me like this is never set and if available the default would be used? 
Do we really need this output?

Reply:
Well we definitely need it as we like to use in different repos dependencies instead of hardcoded api://app for each

regarding the implementation I found much easier way how to retrieve the value based on deeped look into documentation etc ..